source/linux/Foundational_Components_Multimedia_wave5.rst

```diff
@@ -33,17 +32,89 @@ Decoder :
+=====================================================
+
+As shown in the figures below, the software stack of the accelerated
+encoding and decoding has two parts:
+
+.. ifconfig:: CONFIG_part_variant in ('AM62AX', 'AM62PX', 'J722S', 'AM67')
+
+   - A V4L2 (Video4Linux version 2) software driver running on Linux on the A53 MPU subsystem
+   - The firmware running on the DECODER and ENCODER
+
+.. ifconfig:: CONFIG_part_variant in ('J721S2', 'J784S4')
+
+   - A V4L2 (Video4Linux version 2) software driver running on Linux on the A72 MPU subsystem
+   - The firmware running on the DECODER and ENCODER
+
+The driver communicates with the firmware running on the ENCODER/DECODER
+through its own IPC (inter-processor communication).
+
+.. ifconfig:: CONFIG_part_variant in ('AM62AX', 'AM62PX', 'J722S', 'AM67')
+
+   For the DECODER, at the highest level in the MPU subsystem on the A53,
+   there is a Linux user space application which is based on GStreamer. GStreamer
+   is an open source framework that simplifies the development of multimedia
+   applications. The GStreamer library loads and interfaces with the GStreamer
+   plugin (V4L2 plugin), which handles all the details specific to the use of
+   the hardware accelerator. Specifically, the GStreamer plugin interfaces
+   with the V4L2 decoder kernel driver interface.
+
+.. ifconfig:: CONFIG_part_variant in ('J721S2', 'J784S4')
+
+   For the DECODER, at the highest level in the MPU subsystem on the A72,
+   there is a Linux user space application which is based on GStreamer. GStreamer
+   is an open source framework that simplifies the development of multimedia
+   applications. The GStreamer library loads and interfaces with the GStreamer
+   plugin (V4L2 plugin), which handles all the details specific to the use of
+   the hardware accelerator. Specifically, the GStreamer plugin interfaces
+   with the V4L2 decoder kernel driver interface.
+
+.. figure:: ../images/MM_Wave5_SW_overview.png
+   :alt: codec software stack
+
+   CODEC Software Stack
+
+
+Linux Kernel Drivers
+====================
+
+.. rubric:: TI-Provided V4L2 Drivers for Multimedia
+   :name: ti-provided-v4l2-drivers-for-multimedia
+
+Video4Linux version 2 (V4L2) is an open source framework that
+provides a media interface to all Linux-based applications. V4L2 is
+a collection of device drivers and an API for supporting realtime
+video capture and video memory-to-memory operations on Linux systems.
+
+Video encode and decode using the ENCODER and DECODER hardware, respectively,
+are enabled as V4L2 drivers. The V4L2 is integrated with the ENCODER and
+DECODER drivers by a thin layer that implements the V4L2 node ioctls
+and translates the V4L2 data structures to those understood by the
+ENCODER/DECODER.
+
+The V4L2 drivers that are integrated with ENCODER and DECODER are fully conformant
+with the upstream standards. This can be tested as follows:
+
+.. code-block:: console
+
+   v4l2-compliance -d0
 
 .. code-block:: text
```

Comment:
Suggestion:

For larger blocks of text like this, it may be worth splitting it into it's own file, like:
https://github.com/TexasInstruments/processor-sdk-doc/blob/22d26390641fcb436a6ccf3a9e14cfa2d5a36a9f/source/linux/Foundational_Components/Graphics/Rogue/Overview.rst?plain=1#L9-L20
https://github.com/TexasInstruments/processor-sdk-doc/blob/22d26390641fcb436a6ccf3a9e14cfa2d5a36a9f/source/linux/Foundational_Components/Graphics/Rogue/_8XE_Extension_List.rst

Better for platform specific sections and large blocks that are expected to be updated somewhat regularly. Also makes it easier to review changes like this, where the diff itself gets messy - mixing content with the larger block.